mesonbuild/coredata.py

```diff
@@ -595,6 +598,24 @@ def __init__(self, options: argparse.Namespace, scratch_dir: str, meson_command:
         self.builtin_options_libdir_cross_fixup()
         self.init_builtins('')
 
+    @staticmethod
+    def __parse_program_aliases(options: argparse.Namespace, for_machine: MachineChoice) -> OrderedDict[str, str]:
+        alias_map: OrderedDict[str, str] = OrderedDict()
+        aliases: T.List[str]
+        if for_machine == MachineChoice.HOST:
+            aliases = options.native_program or []
+        elif for_machine == MachineChoice.BUILD:
+            aliases = options.cross_program or []
+
+        for a in aliases:
+            parts = a.split('=', 1)
+            if len(parts) == 1:
+                pass
+
+            alias_map[parts[0]] = parts[1]
```

Comment:
There is something broken here. You cannot access parts[1] is len(parts) is 1...

Reply:
There is an `if` guarding the access

Reply:
That should be a continue...